### ISSUE TITLE
Support for RDS IAM Authentication tokens

### ISSUE THREAD

**Issue by samongyr-sq** [cashapp/spirit#285]
Attempting to use `spirit` with an account that uses [RDS IAM DB Authentication](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/UsingWithRDS.IAMDBAuth.html) results in an error like 

```
spirit: error: this user requires clear text authentication. If you still want to use it, please add 'allowCleartextPasswords=1' to your DSN
```

The same error arises when running with `LIBMYSQL_ENABLE_CLEARTEXT_PLUGIN=1`

It would be neat if we could workaround this error with either the env var or a `--enable-cleartext-plugin` equivalent CLI option

**Comment by morgo** [comment]
I took a look at this today. It's going to be a more difficult fix unfortunately :(

- Spirit can be modified easily, but it depends on canal/go-mysql for binlog reading.
- Canal only supports the [basic auth plugins](https://github.com/go-mysql-org/go-mysql/blob/79f480a8090edbd11aa3520e33122749b6519080/client/auth.go#L15-L18).

I modified spirit to try it, and as expected it fails to subscribe to the binlog:
```
ERRO[0013] canal start sync binlog err: start sync replication at binlog (mysql-bin-changelog.000003, 267160) error ERROR 1105 (HY000): Failed to register slave; too long 'report-password'
ERRO[0013] canal has failed. error: start sync replication at binlog (mysql-bin-changelog.000003, 267160) error ERROR 1105 (HY000): Failed to register slave; too long 'report-password'
panic: canal has failed

goroutine 15 [running]:
github.com/cashapp/spirit/pkg/repl.(*Client).startCanal(0x1400047c000)
	/Users/mtocker/go/src/github.com/cashapp/spirit/pkg/repl/client.go:376 +0x234
created by github.com/cashapp/spirit/pkg/repl.(*Client).Run in goroutine 1
	/Users/mtocker/go/src/github.com/cashapp/spirit/pkg/repl/client.go:333 +0x4e0
exit status 2
```

**Comment by samongyr-sq** [comment]
Unfortunately, [it appears that this is a MySQL limitation](https://github.com/go-mysql-org/go-mysql/issues/879#issuecomment-2111418603)

**Comment by morgo** [comment]
This is unfortunate :( I've added the tag won't fix, since we are blocked by MySQL functionality.

If someone sees a way around it, I'm happy to remove the tag.

**Comment by morgo** [comment]
I'm going to close this as won't fix.

**Comment by reillski** [comment]
Hey @morgo looks like gh-ost made the following PR recently. Could it help here?

https://github.com/github/gh-ost/pull/1525
Thanks,
Marc
